salt/kubelet/stop.sls

```diff
@@ -4,6 +4,14 @@
 include:
   - kubectl-config
 
+{% if not "kube-master" in salt['grains.get']('roles', []) %}
+set-should-uncordon-grain:
+  grains.present:
+    - name: kubelet:should_uncordon
+    - value: {{ salt['cmd.run']("kubectl --kubeconfig=" + pillar['paths']['kubeconfig'] + " get nodes " + grains['caasp_fqdn'] + " -o=jsonpath='{.spec.unschedulable}' 2>/dev/null") != "true" }}
+    - force: True
+{% endif %}
+
 # If this fails we should ignore it and proceed anyway as Kubernetes will recover
```

Comment:
If `drain-kubelet` fails in the orchestration for some reason, `kubelet:should_uncordon` will be set anyway (because it was run before `drain-kubelet`). I would do instead:

```yaml

{{ set schedulable = salt['cmd.run']("kubectl --kubeconfig=" + pillar['paths']['kubeconfig'] + " get nodes " + grains['caasp_fqdn'] + " -o=jsonpath='{.spec.unschedulable}' 2>/dev/null") != "true" }} }}

drain-kubelet:
    cmd.run:
     - name: |
         kubectl drain {{ grains['caasp_fqdn'] }} --ignore-daemonsets --grace-period=300 --timeout=340s
     - env:
       - KUBECONFIG: {{ pillar['paths']['kubeconfig'] }}
     - check_cmd:
       - /bin/true
     - require:
       - file: {{ pillar['paths']['kubeconfig'] }}
   grains.present:
     - name: kubelet:should_uncordon
     - value: {{ scheduable }}
     - force: True
     - require:
       - cmd: drain-kubelet
```

Reply:
Good catch, will fix.

Reply:
Taking a look more in depth at your comment I will take the idea, but even if a node was cordoned we still want to drain it, since some workloads could be running there from before the cordoning.